sw/host/opentitanlib/src/image/manifest_def.rs

```diff
@@ -177,9 +177,20 @@ impl ManifestPacked<[ManifestExtTableEntry; CHIP_MANIFEST_EXT_TABLE_COUNT]>
 
     fn overwrite(&mut self, o: Self) {
         for i in 0..self.len() {
-            if !matches!(o[i].0, ManifestExtEntryVar::None) {
-                self[i].0 = o[i].0.clone();
-            }
+            match self[i].0 {
```

Comment:
The logic seems reversed here: in the original, the `other` entry is checked that it is not `None` and then overwrites the corresponding entry in `self`.  In the new version, the value of `self` is checked to see if it should be overwritten with `other`.

It seems like `other` should be checked to see if there is something there to overwrite and then the current values in `self` and `other` should be examined to determine whether or not to perform the overwrite.

Reply:
good point, fixed. Does not look very elegant, but I can't think of a better way.